src/vcModals.cpp

```diff
@@ -619,6 +619,83 @@ void vcModals_DrawProfile(vcState* pProgramState)
   }
 }
 
+void vcModals_DrawChangePassword(vcState *pProgramState)
+{
+  const char *profile = vcString::Get("modalChangePasswordTitle");
+
+  if (pProgramState->openModals & (1 << vcMT_ChangePassword))
+    ImGui::OpenPopup(profile);
+
+  float width = 390.f;
+  float height = 150.f;
+  float buttonWidth = 80.f;
+  ImGui::SetNextWindowSize(ImVec2(width, height), ImGuiCond_Always);
+
+  if (ImGui::BeginPopupModal(profile, nullptr, ImGuiWindowFlags_NoResize))
+  {
+    pProgramState->modalOpen = true;
+
+    if (pProgramState->sessionInfo.isOffline)
+    {
+      ImGui::TextUnformatted(vcString::Get("modalProfileOffline"));
+    }
+    else
+    {
+      ImGuiInputTextFlags passwordFieldFlags = ImGuiInputTextFlags_Password;
+
+      ImGui::InputText(vcString::Get("modalProfileCurrentPassword"), &pProgramState->changePassword.currentPassword[0], vcMaxPathLength, passwordFieldFlags);
+      ImGui::Separator();
+      ImGui::InputText(vcString::Get("modalProfileNewPassword"), &pProgramState->changePassword.newPassword[0], vcMaxPathLength, passwordFieldFlags);
+      ImGui::InputText(vcString::Get("modalProfileReEnterNewPassword"), &pProgramState->changePassword.newPasswordConfirm[0], vcMaxPathLength, passwordFieldFlags);
+    }
+
+    ImGui::Separator();
+    
+    if (ImGui::Button(vcString::Get("modalProfileConfirmNewPassword"), ImVec2(130.0f, 0)) || vcHotkey::IsPressed(vcB_Cancel))
+    {
+      udJSON changePasswordData;
+      changePasswordData.Set("username = \"%s\"", pProgramState->settings.loginInfo.username);
+      changePasswordData.Set("oldpassword = \"%s\"", pProgramState->changePassword.currentPassword);
```

Comment:
If these contain a `"` this will break, you need to use this pattern:
```c++
udJSON temp;
temp.SetString(string);
changePasswordData.Set(&tempNode, "key");
```

Reply:
what's tempNode?

Reply:
This is fixed in latest push. not sure why there's no "outdated" tag on it.

Reply:
The `outdated` tag only applies to lines that are modified.